apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { FormValidation } from "@wso2is/validation";
+import {
+    CommonPluggableComponentMetaPropertyInterface,
+    FederatedAuthenticatorMetaInterface,
+    FederatedAuthenticatorWithMetaInterface
+} from "../../models";
+
+type SamlIdPListItemOption = {
+    key: number;
+    text: string;
+    value: string;
+};
+
+export const getAvailableNameIDFormats = (): Array<SamlIdPListItemOption> => {
+    const schemes: string[] = [
+        "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+        "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",
+        "urn:oasis:names:tc:SAML:2.0:nameid-format:entity",
+        "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+        "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent",
+        "urn:oasis:names:tc:SAML:2.0:nameid-format:encrypted",
+        "urn:oasis:names:tc:SAML:1.1:nameid-format:X509SubjectName",
+        "urn:oasis:names:tc:SAML:1.1:nameid-format:WindowsDomainQualifiedName",
+        "urn:oasis:names:tc:SAML:2.0:nameid-format:kerberos"
+    ];
+    return schemes.map((scheme: string, index: number) => ({
+        key: index, text: scheme, value: scheme
+    }));
+};
+
+export const getAvailableProtocolBindingTypes = (): Array<SamlIdPListItemOption> => {
+    return [
+        { key: 1, text: "HTTP Redirect", value: "redirect" },
+        { key: 2, text: "HTTP Post", value: "post" },
+        { key: 3, text: "As Per Request", value: "as_request" }
+    ];
+};
+
+/**
```

Comment:
Add doc comments for the individual mapping functions and create a issue in `product-is` and link as a TODO.

Reply:
Resolved in commit 4ca4768